hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/runtime/DockerLinuxContainerRuntime.java

```diff
@@ -589,13 +601,17 @@ public void launchContainer(ContainerRuntimeContext ctx)
     Map<String, String> environment = container.getLaunchContext()
         .getEnvironment();
     String imageName = environment.get(ENV_DOCKER_CONTAINER_IMAGE);
+    String selectedImage = environment.get(ENV_DOCKER_SELECTED_CONTAINER_IMAGE);
     String network = environment.get(ENV_DOCKER_CONTAINER_NETWORK);
     String hostname = environment.get(ENV_DOCKER_CONTAINER_HOSTNAME);
     String runtime = environment.get(ENV_DOCKER_CONTAINER_DOCKER_RUNTIME);
     boolean serviceMode = Boolean.parseBoolean(environment.get(
         ENV_DOCKER_CONTAINER_DOCKER_SERVICE_MODE));
     boolean useEntryPoint = serviceMode || checkUseEntryPoint(environment);
 
+    if (selectedImage != null && allowedImages.containsKey(selectedImage)) {
```

Comment:
if selectImage is not null but does not exists, we should probably indicate an error message rather than providing a default image. Otherwise it can mislead users

Reply:
Indeed it would be misleading for users, It's preferable to raise an error here